### ISSUE TITLE
Deploying files on clearing the cache may cause problems with Shell Deployer

### ISSUE THREAD

**Issue by sijones-uk** [putyourlightson/craft-blitz#348]
### Describe the bug
On 30 Dec 2020, as a response to [this issue](https://github.com/putyourlightson/craft-blitz/issues/218), a new line was added to the `clearAll` function in [this line](https://github.com/putyourlightson/craft-blitz/commit/ac97bee9dda37749bec3be3dc98045294f3b8e4e#diff-f129c9cc413d7e49902d8681bbdbdf453bd621ba4294dd63cfeaa56904412df5R95):

```
Blitz::$plugin->deployer->deployAll();
```

Also, the following line was added to the `clearUris` function in [this line](https://github.com/putyourlightson/craft-blitz/commit/ac97bee9dda37749bec3be3dc98045294f3b8e4e#diff-f129c9cc413d7e49902d8681bbdbdf453bd621ba4294dd63cfeaa56904412df5R60):

```
Blitz::$plugin->deployer->deployUris($event->siteUris);
```

This means that when the Clear Cache button is clicked, or the `./craft blitz/cache/clear` command is run, files are not only cleared but also deployed (prior to this commit, files were cleared but not deployed). This means that when using the Shell Deployer, the deployment commands for the remote deployer will run unexpectedly. Also, the refresh command results in multiple deployments occurring, because it runs `refreshAll`, which in turn runs `clearAll` and then `deployUris`.

### To reproduce
Steps to reproduce the behaviour:
1. Install the [Blitz Shell Deployer](https://github.com/putyourlightson/craft-blitz-shell)
1. Go to `/admin/settings/plugins/blitz#deployment` and change "Remote Deployer" to "Shell Deployer". Add a command, eg. `['aws s3 sync ' . CRAFT_BASE_PATH . '/web/uploads s3://' . App::env('S3_BUCKET') . '/uploads']`
2. Go to `/admin/utilities/blitz-cache` and click the "Clear Cache" button
3. Observe that the Blitz cache is cleared, but also that deployment runs as well

### Expected behaviour
When the cache is cleared, deployment should be a configurable option so users can prevent it happening if required.

### Versions
- Plugin version: 1.x-dev
- Craft version: 3.6.18

**Comment by bencroker** [comment]
I agree that https://github.com/putyourlightson/craft-blitz/commit/ac97bee9dda37749bec3be3dc98045294f3b8e4e changed the expected behaviour. If I was to revert it, how would one be expected to deal with the issue as raised in https://github.com/putyourlightson/craft-blitz/issues/218#issue-611430580?

**Comment by sijones-uk** [comment]
Either you could make the deployment calls in those two functions optional, i.e. dependent on a Blitz config setting, or for a more targeted solution, you could revert those changes and listen for the [`DeleteElementEvent`](https://docs.craftcms.com/api/v3/craft-events-deleteelementevent.html) and fire the deployment from that.

**Comment by bencroker** [comment]
Well the thing is that deleting an element triggers a refresh cache job (if the `clearCacheAutomatically` config setting is enabled), which in turn triggers a deploy (if the `warmCacheAutomatically` config setting is enabled). So in theory, everything should "just work". So I think I'll revert the changes introduced in https://github.com/putyourlightson/craft-blitz/commit/ac97bee9dda37749bec3be3dc98045294f3b8e4e. Any thoughts @sijones-uk?

**Comment by sijones-uk** [comment]
That would definitely fix the unexpected behaviour that I'm seeing and would remove the extra overhead on the cache clearing.

**Comment by sijones-uk** [comment]
If you revert that change, please could you create a tagged release for it?

**Comment by bencroker** [comment]
Done in https://github.com/putyourlightson/craft-blitz/commit/8dc32b6711770c97fca48fe2b3a6b40fee6fdcef for the next release, thanks!

**Comment by bencroker** [comment]
Released in version 3.10.4.
